app/controllers/pages_controller.rb

```diff
@@ -0,0 +1,69 @@
+require 'net/http'
+require 'json'
+
+POSTCODE_BASE_API = "https://api.postcodes.io/postcodes/"
+TFL_BASE_API = "https://api.tfl.gov.uk/StopPoint/"
+
+
+
+class PagesController < ApplicationController
+  def index
+  end
+
+  def postcode
+    lat,lon = get_coords_by_postcode
+    if lat == nil && lon == nil
+      flash[:error] = "Invalid postcode. Please try again."
+      redirect_to root_path
+    else
```

Comment:
Maybe try and avoid nested if statement to improve code readability. Maybe return early from the function.